### PR TITLE
Features/error handling

### PR DESCRIPTION
**Pull request by ah584d**
- Handling error on login
 - adding new modal loading/error
 - fixing types and TS rules
- improving debugging options for android

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
   "version": "0.0.1",
   "private": true,
   "scripts": {
-    "android": "react-native run-android",
+    "android": "npm run compile && react-native run-android",
```
Comment (Nirony) [inline package.json]: why did you change it?

File: app/screens/login.tsx [hunk 2]
```diff
@@ -24,17 +31,17 @@ export const LoginScreen: React.FunctionComponent<LoginProps> = observer(props =
...
-	  displayLoginError();
+		let errorMsg = undefined;
+		if(loginReq.status === 401 ) {
+			errorMsg = 'The username or password is incorrect';
+		} else {
+			errorMsg = 'An error occurred when logging in please try again later';
+		}
+		//console.log(`login failed: ${JSON.stringify(loginReq)}`);
+		toggleAlert(/*errorMsg*/);
     }
```
Comment (Nirony) [inline app/screens/login.tsx]: I think its best that this logic should be inside the network request function. 
we want to be able to catch each network error

File: app/screens/login.tsx [hunk 3]
```diff
@@ -66,12 +72,41 @@ export const LoginScreen: React.FunctionComponent<LoginProps> = observer(props =
...
+					</TouchableHighlight> */}
+					<TouchableOpacity onPress={toggleAlert}>
+					<Text>OK</Text>
+				</TouchableOpacity>
+				</FancyAlert>
+			</View>
+		);
+	}
+
   return (
```
Comment (Nirony) [inline app/screens/login.tsx]: i believe that this kind of modal should be placed more globally so it is reusable.

File: package.json (hunk 1 above)
Comment (ah584d) [reply package.json]: in order to launch the last changes in the emulator, i need to compile the application first.
Before this change , i was doing some change in the code, and even reloading the application in the emulator, i was not able to see my changes, until i understood...that the application was not compiled!!
If you prefer i can add a target:
"android": "react-native run-android",
"android:debug": "npm run compile && react-native run-android",

File: app/screens/login.tsx (hunk 2 above)
Comment (ah584d) [reply app/screens/login.tsx]: please be specific, which logic i can't see the updated file here looks to be outdated.
I suppose you speak about 
if(loginResponse.status === 401 ) {
				errorMsg = 'The username or password is incorrect';
			} else {
				errorMsg = 'An error occurred when logging, please try again later';
			}

to move into the login function itself?

File: app/screens/login.tsx (hunk 3 above)
Comment (ah584d) [reply app/screens/login.tsx]: yes i think too! i was not sure, i also want to make a bit order in the ode, i don't know what is the best practice in react-native, i mean i would like to separate the styles in another file, like we do in angular, what is the best practice in react-native ??

File: package.json (hunk 1 above)
Comment (ah584d) [reply package.json]: ok change reverted

File: app/screens/login.tsx (hunk 3 above)
Comment (ah584d) [reply app/screens/login.tsx]: done , i separated the modal and the loading screen into two separated components, that i could customize when i will get the new UI

File: app/screens/login.tsx [hunk 4]
```diff
@@ -2,10 +2,13 @@ import React, { useState } from 'react';
 import { observer } from 'mobx-react-lite';
 import { StyleSheet, View, Alert } from 'react-native';
 import { NavigationActions, NavigationInjectedProps } from 'react-navigation';
-import { Button, Checkbox, Icon, Screen, TextField } from '../components';
+import { Button, Checkbox, Icon, Screen, TextField, Text } from '../components';
 import { color, spacing } from '../theme';
 import { Toggle } from 'react-powerplug';
 import { useStores } from '../models/root-store';
+import { FancyAlert } from 'react-native-expo-fancy-alerts';
```
[the remark this answers is not in the document]
Comment (ah584d) [reply app/screens/login.tsx]: i deleted the dependency, i implemented it by myself

File: app/screens/login.tsx (hunk 2 above)
Comment (ah584d) [reply app/screens/login.tsx]: will be done in next PR, i will cut the modal logic from the component

File: package.json [hunk 5]
```diff
@@ -3,7 +3,7 @@
   "version": "0.0.1",
   "private": true,
   "scripts": {
-    "android": "npm run compile && react-native run-android -- --reset-cache",
+    "android": "react-native run-android -- --reset-cache",
```
Comment (Nirony) [inline package.json:6]: please remove --reset-cache flag

File: app/screens/login.tsx [hunk 6]
```diff
@@ -2,20 +2,24 @@ import React from 'react';
...
 import { color, spacing } from '../theme';
 import { Toggle } from 'react-powerplug';
 import { useStores } from '../models/root-store';
-import { FancyAlert, LoadingIndicator } from 'react-native-expo-fancy-alerts';
+import { FancyAlert } from 'react-native-expo-fancy-alerts';
 import VectorIcon from 'react-native-vector-icons/FontAwesome';
 import PromiseTimeout from '../services/timers/promise-timeout.service';
+import { Button, Checkbox, Screen, Icon, TextField } from '../components';
+import ErrorModal from '../components/modal-error/error.modal';
+import LoadingModal from '../components/loading/loading.modal';
```
Comment (Nirony) [inline app/screens/login.tsx:12]: side note, try not to use dotes in file name in react...

File: app/components/modal-error/error.modal.tsx [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+		// if (prevState.isVisible !== nextProps.visible) {
+		//   return {
+		// 	isVisible: nextProps.visible
+		//   };
+		// }
+	
+		// Return null to indicate no change to state.
+		return null;
+	}
+
```
Comment (Nirony) [inline app/components/modal-error/error.modal.tsx]: comments are necessary here?

File: app/components/modal-error/error.modal.tsx [hunk 8]
```diff
@@ -0,0 +1,168 @@
+import React, { Component } from 'react';
+import { Modal, ActivityIndicator, View, StyleSheet, Dimensions, TouchableOpacity} from 'react-native';
+import VectorIcon from 'react-native-vector-icons/FontAwesome';
+import { color } from '../../theme';
+import { Text } from '../text/text';
+const { width: SCREEN_WIDTH } = Dimensions.get('window');
+const ALERT_WIDTH = SCREEN_WIDTH - (SCREEN_WIDTH / 4);
+
```
Comment (Nirony) [inline app/components/modal-error/error.modal.tsx:8]: you could also set the width to 25%

File: app/components/modal-error/error.modal.tsx [hunk 9]
```diff
@@ -0,0 +1,168 @@
...
+	 }
+	state = {
+		isVisible: false
+	 }
+	 toggleModal(show: boolean) {
+		this.setState({ isVisible: show });
+	 }
+   render() {
+      return (
+		<Modal visible={this.state.isVisible} animationType="fade" transparent onRequestClose={() => null }>
```
Comment (Nirony) [inline app/components/modal-error/error.modal.tsx]: i think eslint annotation setting is a bit too much... up to you :)

**Comment by ah584d** [comment]
too much comments
